include/boost/beast/http/impl/read.ipp

```diff
@@ -101,6 +115,12 @@ public:
         asio_handler_invoke(
             f, std::addressof(op->h_));
     }
+
+    void do_upcall(error_code ec) const
+    {
+        s_.get_io_service().post(
+            bind_handler(std::move(*this), ec, 0));
```

Comment:
I prefer to do this explicitly inside the main function rather than as a separate function. Also I would not mark it `const`. Beast does not interpret `const` as a "bitwise const" as you have done here. Instead, `const` is roughly interpreted as "thread safe" - which this function is not.